packages/api/src/utils/authenticateUser.ts

```diff
@@ -0,0 +1,25 @@
+import { Response , Request } from 'express';
+
+type OAuthUserData = {
+    email: string ,
+    firstName?: string ,
+    lastName?: string ,
+    phoneNumber?: string ,
+    address?: string ,
+    given_name?: string ,
+    family_name?: string ,
```

Comment:
First/last should be necessary for everything and should be accessible via the APIs for each auth provider so let's make those required. Beyond that, I'd prefer to start with just what we need and expand from there
```suggestion
    firstName: string ,
    lastName: string ,
```